src/Autowhatever.js

```diff
@@ -281,7 +283,7 @@ export default class Autowhatever extends Component {
 
     return (
       <div {...theme(`react-autowhatever-${id}-container`, 'container', isOpen && 'containerOpen')}>
-        <input {...inputProps} />
+        {this.props.renderInput(inputProps)}
```

Comment:
it's not the same `inputProps` that's passed in, it's the one that has defaults and other things merged in. I suppose the alternative would be to simply expose an `inputElement` prop defaulting to `'input'`, that way one can just pass a React component.